controllers/openstackdataplanenodeset_controller.go

```diff
@@ -193,6 +199,20 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 
 	instance.Status.Conditions.MarkFalse(dataplanev1.SetupReadyCondition, condition.RequestedReason, condition.SeverityInfo, condition.ReadyInitMessage)
 
+	// Detect config changes and set Status ConfigHash
+	configHash, err := r.GetSpecConfigHash(instance)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if configHash != instance.Status.DeployedConfigHash {
+		instance.Status.ConfigHash = configHash
+		err := helper.PatchInstance(ctx, instance)
```

Comment:
isn't this a redundant call to PatchInstance given that happens in `defer func`? If you drop this call does instance.Status.ConfigHash still get updated?

Reply:
Hmm, yeah, I think you're right. Because even if the reconcile fails for any other reason, it will run the defer func(). So we should still end up with the same result. Let's drop it and see how it goes.